src/gstreamer_cam.cpp

```diff
@@ -0,0 +1,171 @@
+#include "gstreamer_cam.hpp"
+
+GStreamerCam::GStreamerCam(ros::NodeHandle &nh)
+{
+    nh_ = nh;
+    nh_.getParam("camera_location", camera_location_);
+    nh_.getParam("camera_format", camera_format_);
+    nh_.getParam("camera_width", camera_width_);
+    nh_.getParam("camera_height", camera_height_);
+    nh_.getParam("camera_fps", camera_fps_);
+    nh_.getParam("camera_topic", camera_topic_);
+
+    rosImagePub_ = nh_.advertise<sensor_msgs::Image>("/camera/image_rect", 1);
+    rosCameraInfoPub_ = nh_.advertise<sensor_msgs::CameraInfo>("/camera/camera_info", 1);
+    
+    // init the gstreamer pipeline
+    InitializeGStreamer(); 
+    SetCameraParams(); // set distortions and camera info (not fully implemented yet)
```

Comment:
What is your proposed implementation? Would we be reading this from a yaml file specified in the params or something like that?

Reply:
yeah think reading from a yaml would be the easiest. Do you know what the camera calibration ros package outputs? We can structure it that way

Reply:
I'm not sure where this is documented officially (because there seems to be several formats, including an ini format, a YAML format that OpenCV uses, and whatever camera calibration outputs), but it's shown [here](https://answers.ros.org/question/248563/publishing-camerainfo-messages-from-yaml-file/) (the snippet in the second answer).

You can also check the `sae_fs` code because I'm pretty sure we did this exact thing last year for all the projection stuff. I'm pretty sure we used the exact same code snippet as the one shown here actually.